example-purescript-package/spago2.dhall

```diff
@@ -0,0 +1,6 @@
+{ name = "my-project"
+, dependencies = [ "console", "effect", "prelude", "psci-support" ]
+, packages = https://github.com/purescript/package-sets/releases/download/psc-0.14.4-20210905/packages.dhall sha256:140f3630801f2b02d5f3a405d4872e0af317e4ef187016a6b00f97d59d6275c6
+, sources = [ "src/**/*.purs", "test/**/*.purs" ]
+}
```

Comment:
This file is an example of my motivation for writing this PR.

I'd like to be able to package PureScript packages in Nix.  The user should only need this `spago.dhall` (and the accompanying `packages.dhall` file, but I've inlined `packages.dhall` in this `spago2.dhall` file).

In PureScript, the build tool `spago` is setup to always add an integrity check to the package set import, so in theory this should be possible to build with a FOD nix derivation.  This PR implements the FOD fetcher.  I still have to put together an example of actually building a PureScript application with this.